arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Sequence.kt

```diff
@@ -610,15 +611,37 @@ public fun <A, B> Sequence<Validated<A, B>>.separateValidated(): Pair<Sequence<A
   return asep to bsep
 }
 
+public fun <E, A> Sequence<Either<E, A>>.sequence(): Either<E, List<A>> =
+  traverse(::identity)
+
+@Deprecated(
+  "sequenceEither is being renamed to sequence to simplify the Arrow API",
+  ReplaceWith("sequence().map { it.asSequence() }", "arrow.core.sequence")
+)
 public fun <E, A> Sequence<Either<E, A>>.sequenceEither(): Either<E, Sequence<A>> =
-  traverseEither(::identity)
+  sequence().map { it.asSequence() }
+
+public fun <A> Sequence<Option<A>>.sequence(): Option<List<A>> =
+  traverse(::identity)
 
+@Deprecated(
+  "sequenceOption is being renamed to sequence to simplify the Arrow API",
+  ReplaceWith("sequence().map { it.asSequence() }", "arrow.core.sequence")
+)
 public fun <A> Sequence<Option<A>>.sequenceOption(): Option<Sequence<A>> =
-  traverseOption(::identity)
+  sequence().map { it.asSequence() }
+
+public fun <E, A> Sequence<Validated<E, A>>.sequence(semigroup: Semigroup<E>): Validated<E, List<A>> =
+  traverse(semigroup, ::identity)
 
+@Deprecated(
+  "sequenceValidated is being renamed to sequence to simplify the Arrow API",
+  ReplaceWith("sequence(semigroup).map { it.asSequence() }", "arrow.core.sequence")
+)
 public fun <E, A> Sequence<Validated<E, A>>.sequenceValidated(semigroup: Semigroup<E>): Validated<E, Sequence<A>> =
-  traverseValidated(semigroup, ::identity)
+  sequence(semigroup).map { it.asSequence() }
 
+@Deprecated("Deprecated legacy Api", ReplaceWith("map { generateSequence { this } }"))
```

Comment:
```suggestion
@Deprecated("some is being deprecated in favor of map", ReplaceWith("map { generateSequence { this } }"))
```
Is this message better @nomisRev ?

Reply:
Yes, that is great 👍  I was more concerned about the APIs we're renaming but not removing.